server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/managable/cassandra/DeletedMailsDaoTest.java

```diff
@@ -0,0 +1,145 @@
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.managable.cassandra;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Optional;
+
+import javax.mail.MessagingException;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.backends.cassandra.DockerCassandraRule;
+import org.apache.james.queue.api.ManageableMailQueue;
+import org.apache.james.queue.rabbitmq.MailQueueName;
+import org.apache.james.queue.rabbitmq.managable.cassandra.model.CassandraManageableMailQueueItem;
+import org.apache.james.queue.rabbitmq.managable.cassandra.model.MailKey;
+import org.apache.mailet.base.test.FakeMail;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class DeletedMailsDaoTest {
+
+    private static CassandraManageableMailQueueItem createQueueItem(MailQueueName mailQueueName, MailKey mailKey) throws MessagingException {
+        return new CassandraManageableMailQueueItem(
+            mailQueueName,
+            mailKey,
+            new ManageableMailQueue.MailQueueItemView(FakeMail.builder()
+                .name(mailKey.getMailKey())
+                .build()));
+    }
+
+    private static final MailQueueName INBOX_1 = MailQueueName.fromString("INBOX_1");
+    private static final MailQueueName INBOX_2 = MailQueueName.fromString("INBOX_2");
+    private static final MailKey MAIL_KEY_1 = MailKey.of("mailkey1");
+    private static final MailKey MAIL_KEY_2 = MailKey.of("mailkey2");
+
+    @ClassRule
+    public static DockerCassandraRule cassandraServer = new DockerCassandraRule();
+    private static CassandraCluster cassandra;
+
+    private DeletedMailsDao testee;
+
+    @BeforeClass
+    public static void setUpClass() {
+        cassandra = CassandraCluster.create(CassandraManageableMailQueueModule.MODULE, cassandraServer.getHost());
+    }
+
+    @Before
+    public void setUp() {
+        testee = new DeletedMailsDao(cassandra.getConf());
+    }
+
+    @After
+    public void tearDown() {
+        cassandra.clearTables();
+    }
+
+    @AfterClass
+    public static void tearDownClass() {
+        cassandra.closeCluster();
+    }
+
+    @Test
+    public void filterNotDeletedShouldReturnItemWhenTableIsEmpty() throws MessagingException {
+        CassandraManageableMailQueueItem manageableMailQueueItem = createQueueItem(INBOX_1, MAIL_KEY_1);
+        Optional<CassandraManageableMailQueueItem> maybeQueueItem = testee
+            .filterNotDeleted(manageableMailQueueItem)
```

Comment:
I'd argue that DAO should not be filtering data.

It should just answer the question "was it deleted?" and let the caller do it's filtering if he needs.